src/v/model/transform.h

```diff
@@ -170,6 +173,10 @@ struct transform_metadata_patch {
     std::optional<absl::flat_hash_map<ss::sstring, ss::sstring>> env;
     // Desired paused state for the transform
     std::optional<is_transform_paused> paused;
+    // Desired compression mode for the transform
+    std::optional<compression> compression_mode;
```

Comment:
what's the difference between compression_mode:none and nullopt here?

Reply:
in this context, whether or not to update the setting on persisted metadata. comments on these fields could be better.